### ISSUE TITLE
[coq] -R flag by default

### ISSUE THREAD

**Issue by Lysxia** [ocaml/dune#2585]
Currently Coq libraries are prefixed via the "recursive" `-R` flag, rather than "nonrecursive" `-Q` https://github.com/ocaml/dune/blob/master/doc/dune-files.rst#coqtheory

Is there a reason for that choice? Can it be made configurable? I would prefer `-Q` to be the default, because implicitly guessing packages where modules come from (which `-R` enables) sounds like a bad idea.

**Comment by ejgallego** [comment]
@Lysxia `-R` is used only for the current library, all imports are processed with `-Q`. I think we can add a `wrapped` flag tho to even make this the default for the current library.

**Comment by ejgallego** [comment]
Ouch, I just realized that the branch I'm using [which supports a `theories` field] has still not landed on master, sorry for the confusion. Indeed, this will be granted once #2053 lands.

**Comment by ejgallego** [comment]
Yes, that's the idea, however a bit of care is needed w.r.t. compatibility due to Coq still binding by default the global `user-contribs` directory [version 2 of the Coq Dune language will fix that, but break compat with non-Dune built libraries]

**Comment by Blaisorblade** [comment]
Status update: #2053 is in, this issue is still open, and Emilio pointed to https://github.com/ocaml/dune/issues/3314 on [a Zulip thread](https://coq.zulipchat.com/#narrow/stream/240550-Dune-devs.20.26.20users/topic/dune.20and.20coqc.20-R.20vs.20-Q).

**Comment by ejgallego** [comment]
Yeah we can still have libraries to have to qualify other modules from themselves if we'd like to, tho it still seems to me that the default [making libraries access their own modules without prefix] is more flexible.

Note that in my opinion we shouldn't see this discussion in terms of Coq's low-level flags, but more in terms of how we'd like things to work. For example using `-R` should indeed work differently with recursive paths, however there are wider issues upstream due to the semantics of `From`, that for example prevent coqdep / etc... of working without access to the full filesystem layout, _à la_ `ocamldep -modules`.

From my part, more work on this has been pending on unifying the code for handling `-R/-Q` and loadpaths in Coq itself, as of today there are several copies of that code in the codebase so that's a pain to work on.

**Comment by Alizter** [comment]
@ejgallego Update on this issue? What should we do about `-R`? Today we do `-Q` for theories we depend on and `-R ` for the current theory, which seems to be the most sensible option / good practice.

**Comment by ejgallego** [comment]
Yes, I think it seems the most sensible setup, however we have a pending discussion on how we'd like to move from `user-contrib` to a better model, so I guess at some point we should start a design document on what model we'd like to have for `(lang coq 2.0)`

**Comment by Alizter** [comment]
We need to have a proper environment variable instead of relying on user-contrib (Coq side) IMO.
